### PR TITLE
[catnap] Bad `OperationResult` for `accept()`

### PR DESCRIPTION
**Pull request by ppenna**
Description
========

This pull request fixes issue https://github.com/demikernel/demikernel/issues/58.

### CODE REVIEW

File: src/catnap/futures/accept.rs [hunk 1]
```diff
@@ -33,6 +33,8 @@ pub struct AcceptFuture {
     qd: QDesc,
     /// Underlying file descriptor.
     fd: RawFd,
+    /// Queue descriptor of incoming connection
```
Comment (BrianZill) [inline src/catnap/futures/accept.rs:36]: This comment is missing a period.

Comment (ppenna) [reply src/catnap/futures/accept.rs]: Sure.

File: src/catnap/mod.rs [hunk 2]
```diff
@@ -413,7 +414,23 @@ impl CatnapLibOS {
...
-        boxed_concrete_type.get_result()
+
+        let (qd, new_qd, new_fd, qr): (QDesc, Option<QDesc>, Option<RawFd>, OperationResult) =
+            boxed_concrete_type.get_result();
+
+        // Handle accept operation.
+        if let Some(new_qd) = new_qd {
+            // Associate raw file descriptor with queue descriptor.
+            if let Some(new_fd) = new_fd {
+                assert_eq!(self.sockets.insert(new_qd, new_fd).is_none(), true);
```
Comment (BrianZill) [inline src/catnap/mod.rs:425]: Shouldn't this be "assert!(self.sockets.insert(new_qd, new_fd).is_none());" ?

File: src/catnap/mod.rs [hunk 3]
```diff
@@ -413,7 +414,23 @@ impl CatnapLibOS {
...
+            }
+            // Release entry in queue table.
```
Comment (BrianZill) [inline src/catnap/mod.rs:427]: This is a cleanup after some failure?

File: src/catnap/mod.rs (hunk 2 above)
Comment (ppenna) [reply src/catnap/mod.rs:425]: Both of these asserts would be fine, but I'm happy to change it to the way you've suggested.

File: src/catnap/mod.rs (hunk 3 above)
Comment (ppenna) [reply src/catnap/mod.rs:427]: Yes it is. I'm gonna move the comment to the line bellow.

**Review by BrianZill — APPROVED** [review]
Looks good now.  :-)

File: src/catnap/mod.rs (hunk 2 above)
Comment (BrianZill) [reply src/catnap/mod.rs]: Yes, it was just a recommended style thing.

**Review by BrianZill — APPROVED** [review]

**Pull request merged**